File: Array/littleEliphant&Candies.cpp
#include<iostream>
using namespace std;

int main()
{

    int choco,tests,elephants;

    long long chocholates,sum;


    cin>>tests;

    for(int i=0; i<tests; i++)
    {

        sum=0;

        cin>>elephants>>chocholates;

        for(int j=0; j<elephants; j++)
        {
            cin>>choco;
            sum+=choco;
        }
       if(sum<=chocholates)
        cout<<"Yes"<<endl;

    else
        cout<<"No"<<endl;
    }

}

File: Array/littleEliphant&Candies.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Dec 15 11:20:53 2020

@author: gourob
"""

test = int(input())


for i in range(test):
    
    elephant,chocolates = map(int,input().split())
    choco = list(map(int,input().split()))
        
    if sum(choco) <= chocolates:
        print("Yes")
    else:
        print("No")
 